webapp/src/components/ProgramActivityDialog.tsx

```diff
@@ -1,85 +1,212 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { decodeHTML } from 'entities';
+import CancelIcon from '@mui/icons-material/Cancel';
+import CloseIcon from '@mui/icons-material/Close';
+import TaskAltIcon from '@mui/icons-material/TaskAlt';
+import LoadingButton from '@mui/lab/LoadingButton';
 import {
-  Button,
+  Alert,
   Dialog,
   DialogActions,
   DialogContent,
   DialogContentText,
   DialogTitle,
   Divider,
+  FormGroup,
+  Snackbar,
   Table,
   TableBody,
   TableCell,
   TableContainer,
   TableRow,
 } from '@mui/material';
-
+import IconButton from '@mui/material/IconButton';
 import { formatDate, formatTime } from '../helpers/dateTime';
 import { CalendarEvent } from '../types/api';
-
 interface ProgramActivityDialogProps {
   show: boolean;
   contents: CalendarEvent;
   handleClose: () => void;
 }
+const sendAPIGetRequest = (
+  path: string,
+  activityType: string,
+  setCompleted: React.Dispatch<React.SetStateAction<boolean | null>>,
+  setError: React.Dispatch<React.SetStateAction<boolean | null>>,
+  setIsErrorShown: React.Dispatch<React.SetStateAction<boolean>>,
+  setIsLoading: React.Dispatch<React.SetStateAction<boolean>>
+) => {
+  setIsLoading(true);
+  if (activityType !== 'assignment') return;
+  return fetch(`${process.env.REACT_APP_API_ORIGIN}${path}`, {
+    method: 'GET',
+    credentials: 'include',
+  })
+    .then(res => res.json())
+    .then(
+      async ({ data }) => {
+        if (data) {
+          setIsLoading(false);
+          setError(false);
+          setCompleted(data.completed);
+        }
+      },
+      error => {
+        setIsLoading(false);
```

Comment:
In `useApiData()`, we handle ignoring an AbortError, so maybe we should have that exception here. Also, it'd be great if we could pass the error message back to the toast so we can be more specific about the error we encountered, but it's not required within the scope of this ticket.